index/server/main.go

```diff
@@ -149,12 +151,29 @@ func main() {
 
 	router.GET("/index/:type", func(c *gin.Context) {
 		indexType := c.Param("type")
+		iconType := c.Query("icon")
 
 		// Serve the index with type
 		if indexType == string(indexSchema.SampleDevfileType) {
-			c.File(sampleIndexPath)
+			if iconType == "base64" {
+				bytes, err := encodeIndexIconToBase64(sampleIndexPath)
+				if err != nil {
+					log.Fatal(err.Error())
+				}
+				c.Data(http.StatusOK, http.DetectContentType(bytes), bytes)
+			} else {
+				c.File(sampleIndexPath)
+			}
```

Comment:
I would almost like to say if its a wrong param we should 404

Reply:
Or 400 bad request maybe?

Reply:
Probably 400 is more generic and suitable here, IMO URL like `/students/{student ID}` and if the student ID is not in the server then we should return 404.